### ISSUE TITLE
$this-> , autocomplete is not work

### ISSUE THREAD

**Issue by akoweb** [lojadev5/opencart-phpstorm#1]
hi dear , first very thanks for create this extension.
in phpstorm 2019.2
after import extension files , $this-> , autocomplete is not work

**Comment by Daisuke-sama** [comment]
hey @akoweb 
it works for me after writing a property of $this. The root controller is initialized with magical properties that is why you don't have the first level of autocompletion.

**Comment by andrey-reznik** [comment]
Same problem

**Comment by andrey-reznik** [comment]
I am found a solution, need to add a PHPDoc comment in file `system/engine/controller.php` after `<?php` string. It should turn out like this:
```
<?php

/**
 * @mixin OpenCart
 */

abstract class Controller {
	protected $registry;

    public function __construct($registry) {
		$this->registry = $registry;
	}

	public function __get($key) {
		return $this->registry->get($key);
	}

	public function __set($key, $value) {
		$this->registry->set($key, $value);
	}
}
```
And then everything should works fine.

**Comment by lojadev5** [comment]
Fixed on release 1.2
